pkgs/desktops/gnome-3/extensions/pop-os-shell/default.nix

```diff
@@ -0,0 +1,38 @@
+{ stdenv, fetchFromGitHub, nodePackages, glib, substituteAll, gjs }:
+
+stdenv.mkDerivation rec {
+  pname = "pop-os-shell";
+  version = "1.1.0";
+
+  src = fetchFromGitHub {
+    owner = "pop-os";
+    repo = "shell";
+    rev = version;
+    sha256 = "sha256-KOp/0R7P/iH52njr7JPDKd4fAoN88d/pfou2gWy5QPk=";
+  };
+
+  nativeBuildInputs = [ glib nodePackages.typescript ];
+
```

Comment:
You will still need to add `buildInputs = [ gjs ];` so that `patchShebangs` can find it and substitute it.